src/xcommon.cpp

```diff
@@ -186,6 +197,69 @@ namespace xw
         m_comm.close(nl::json::object(), nl::json::object(), xeus::buffer_sequence());
     }
 
+    namespace
+    {
+        std::string tolower(std::string s)
+        {
+            auto safe_tolower = [](unsigned char c)
+            {
+                return std::tolower(c);
+            };
+            std::transform(s.begin(), s.end(), s.begin(), safe_tolower);
+            return s;
+        }
+
+        std::string ltrim(std::string s)
+        {
+            auto const safe_isnotspace = [](unsigned char ch)
+            {
+                return !std::isspace(ch);
+            };
+            s.erase(s.begin(), std::find_if(s.begin(), s.end(), safe_isnotspace));
+            return s;
+        }
+
+        std::string rtrim(std::string s)
+        {
+            auto const safe_isnotspace = [](unsigned char ch)
+            {
+                return !std::isspace(ch);
+            };
+            s.erase(std::find_if(s.rbegin(), s.rend(), safe_isnotspace).base(), s.end());
+            return s;
+        }
+
+        std::string trim(std::string s)
+        {
+            rtrim(s);
+            ltrim(s);
+            return s;
+        }
+
+        bool is_true_string(const char* str)
+        {
+            const std::string s = tolower(trim(str));
+            static const auto trues = {"true", "on", "yes", "1"};
+            return std::find(trues.begin(), trues.end(), s) < trues.end();
+        }
+
+        xtl::xoptional<bool> get_tristate_env(const char* name)
+        {
+            const char* const val = std::getenv(name);
+            if (val == nullptr)
+            {
+                return {};
+            }
+            return is_true_string(val);
+        }
+    }
+
+    xtl::xoptional<bool> xcommon::global_echo_update()
+    {
+        static const auto out = get_tristate_env("JUPYTER_WIDGETS_ECHO");
```

Comment:
Why do we need to check "on", "true", "yes", "1"?

Reply:
I think that was only for ipywidgets 7, where we had to opt in for backwards compatibility . I think in ipywidgets 8 there is no op in or out anymore. But not 100% sure.

Reply:
I was wrong, it's still in, and the bar for accepting values was set quite high: https://github.com/jupyter-widgets/ipywidgets/blob/main/python/ipywidgets/ipywidgets/widgets/widget.py#L29 :)

Reply:
Wow, thanks